src/Hazelcast.Net/Clustering/ClusterState.cs

```diff
@@ -429,7 +431,17 @@ public void RequestShutdown()
         /// <summary>
         /// Whether smart routing is enabled.
         /// </summary>
-        public bool IsSmartRouting => Options.Networking.SmartRouting;
+        public bool IsSmartRouting => RoutingMode == RoutingModes.AllMembers;
```

Comment:
The Java client has dropped all use of "smart routing" except in the client config option, which is maintained for backwards compatibility only - I'd suggest we do the same for C#

This applies to other areas of the PR too, but I will only mention it here.

Reply:
Fair enough. Refactored that smart routing stays same by setting the `RoutingMode`